src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -935,24 +961,28 @@ public RedisSentinelConnection getSentinelConnection() {
 
 	private Jedis getActiveSentinel() {
 
-		Assert.isTrue(RedisConfiguration.isSentinelConfiguration(configuration), "SentinelConfig must not be null");
-		SentinelConfiguration sentinelConfiguration = (SentinelConfiguration) configuration;
+		Assert.isTrue(RedisConfiguration.isSentinelConfiguration(this.configuration),
+				"SentinelConfig must not be null");
+
+		SentinelConfiguration sentinelConfiguration = (SentinelConfiguration) this.configuration;
 
 		JedisClientConfig clientConfig = createSentinelClientConfig(sentinelConfiguration);
+
 		for (RedisNode node : sentinelConfiguration.getSentinels()) {
 
 			Jedis jedis = null;
 			boolean success = false;
 
 			try {
 
-				jedis = new Jedis(new HostAndPort(node.getHost(), node.getPort()), clientConfig);
+				jedis = createJedis(node, clientConfig);
+
 				if (jedis.ping().equalsIgnoreCase("pong")) {
 					success = true;
 					return jedis;
 				}
-			} catch (Exception ex) {
-				log.warn(String.format("Ping failed for sentinel host: %s", node.getHost()), ex);
+			} catch (Exception cause) {
+				log.warn("Ping failed for sentinel host: %s".formatted(node.getHost()), cause);
```

Comment:
Please let's remain on `String.format(…)` to remain consitent. We can change all occurrences of `String.format(…)` to `formatted` if we feel that the value we get from that outweighs the effort.

Reply:
`"some string with placeholders".formatted(..)` is less noisy

Reply:
Case in point: https://github.com/spring-projects/spring-data-redis/blob/3.2.0-RC1/src/main/java/org/springframework/data/redis/core/script/DigestUtils.java#L63